x/oracle/keeper/task.go

```diff
@@ -31,15 +31,21 @@ func (k Keeper) DeleteTask(ctx sdk.Context, task types.TaskI) error {
 }
 
 // UpdateAndSetTask updates a task and set it in KVStore.
-func (k Keeper) UpdateAndSetTask(ctx sdk.Context, task types.TaskI) {
+func (k Keeper) UpdateAndSetTask(ctx sdk.Context, task *types.Task) {
 	if task.IsValid(ctx) {
 		k.SetClosingBlockStore(ctx, task)
 	}
-	if scTask, ok := task.(*types.Task); ok {
-		scTask.ExpireHeight = ctx.BlockHeight() + scTask.WaitingBlocks
-		k.SetTask(ctx, scTask)
-	} else {
+
+	task.ExpireHeight = ctx.BlockHeight() + task.WaitingBlocks
+	k.SetTask(ctx, task)
+}
+
+func (k Keeper) SetTxTask(ctx sdk.Context, task *types.TxTask) {
+	if !task.Expiration.Before(ctx.BlockTime()) {
 		k.SetTask(ctx, task)
```

Comment:
the TxTask should be added to `ExpireTaskStore`.
Please refer to branch `precog-respond` for details, or maybe you could do it after that branch is merged